test/brain/index_selection_test.cpp

```diff
@@ -0,0 +1,330 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// index_selection_test.cpp
+//
+// Identification: test/brain/index_selection_test.cpp
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include <numeric>
+
+#include "binder/bind_node_visitor.h"
+#include "brain/index_selection.h"
+#include "brain/index_selection_util.h"
+#include "brain/what_if_index.h"
+#include "catalog/index_catalog.h"
+#include "common/harness.h"
+#include "concurrency/transaction_manager_factory.h"
+#include "optimizer/stats/column_stats.h"
+#include "optimizer/stats/stats_storage.h"
+#include "optimizer/stats/table_stats.h"
+#include "sql/testing_sql_util.h"
+
+namespace peloton {
+namespace test {
+
+//===--------------------------------------------------------------------===//
+// IndexSelectionTest
+//===--------------------------------------------------------------------===//
+
+class IndexSelectionTest : public PelotonTest {
+ public:
+  IndexSelectionTest() {}
+
+  // Create a new database
+  void CreateDatabase(std::string db_name) {
```

Comment:
Done

Reply:
It might make sense to later setup a Index selection testing utility - a set of simple and more complex workloads with expected indexes that should be selected.